frontend/packages/ceph-storage-plugin/src/components/independent-mode/install.tsx

```diff
@@ -0,0 +1,266 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { match } from 'react-router';
+import {
+  ButtonBar,
+  NsDropdown,
+  withHandlePromise,
+  HandlePromiseProps,
+} from '@console/internal/components/utils';
+import { k8sGet, k8sCreate, referenceForModel } from '@console/internal/module/k8s';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager';
+import { Title, FormGroup, Form, ActionGroup, Button, TextInput } from '@patternfly/react-core';
+import { history } from '@console/internal/components/utils/router';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { getName } from '@console/shared';
+import { OCSServiceModel } from '../../models';
+import FileUpload from './fileUpload';
+import { DataState, ErrorType, Field } from './types';
+import { getValidJSON, checkError } from './utils';
+import './install.scss';
+
+const ERROR: DataState = {
+  ns: '',
+  fsid: '',
+  admin: '',
+  mondata: '',
+};
+
+type ErrorTextProps = {
+  text: string;
+};
+
+const ErrorText: React.FC<ErrorTextProps> = ({ text }) => (
+  <span className="im-install-page--error">{text}</span>
+);
+
+const getErrorText = (text: string) => <ErrorText text={text} />;
+
+const InstallExternalCluster = withHandlePromise((props: InstallExternalClusterProps) => {
+  const {
+    inProgress,
+    errorMessage,
+    handlePromise,
+    match: {
+      params: { ns, appName },
+    },
+  } = props;
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+  const [fileData, setFileData] = React.useState('');
+  const [fileName, setFileName] = React.useState('');
+  const [namespace, setNamespace] = React.useState(ns);
+  const [externalFSID, setExternalFSID] = React.useState(null);
+  const [externalAdminSecret, setExternalAdminSecret] = React.useState(null);
+  const [externalMonData, setExternalMonData] = React.useState(null);
+  const [dataError, setDataError] = React.useState(ERROR);
+  const [fileError, setFileError] = React.useState('');
+  const [, updateState] = React.useState();
+
+  // Todo(bipuladh): React does shallow comparison dataError and fileError need deep comparison.
+  const forceUpdate = React.useCallback(() => updateState({}), []);
+
+  const setErrors = React.useCallback((errors: ErrorType[]): void => {
```

Comment:
We are not passing any argument to callback. I think we can create a normal function.
```suggestion
  const setErrors = React.useCallback((errors: ErrorType[]): void => {
```

Reply:
If I am not mistaken I did this so that I can use this in the dependency array of useEffect.

Reply:
But is it required as you are anyway calling the function inside the useEffect.